File: LENGUAJES/JAVASCRIPT/Ejercicio2.js
// Bucles WHILE

function obtenerNumero() {
    while (true) {
        let entrada = prompt("Ingrese un número para conocer la tabla de multiplicar:");
        let num = parseInt(entrada);
        
        if (!isNaN(num)) {
            return num;
        } else {
            alert("Error, por favor ingrese un número entero.");
        }
    }
}

function mostrarTabla(num) {
    console.log(`\nTabla de multiplicar del ${num}:`);
    for (let i = 1; i <= 10; i++) {
        console.log(`${num} x ${i} = ${num * i}`);
    }
}

function main() {
    while (true) {
        let numero = obtenerNumero();
        mostrarTabla(numero);
        
        let continuar = prompt("\n¿Desea calcular otra tabla? (s/n):").trim().toLowerCase();
        if (continuar !== 's') {
            console.log("Gracias por usar el programa. ¡Hasta luego!");
            break;
        }
    }
}
main();

File: LENGUAJES/GO/Ejercicio2.go
//BUCLES WHILE

package main

import (
	"fmt"
	"strconv"
)

func main() {
	for {
		numero := obtenerNumero()
		mostrarTabla(numero)

		var continuar string
		fmt.Print("\n¿Desea calcular otra tabla? (s/n): ")
		fmt.Scanln(&continuar)
		if continuar != "s" {
			fmt.Println("Gracias por usar el programa. ¡Hasta luego!")
			break
		}
	}
}

func obtenerNumero() int {
	for {
		fmt.Print("Ingrese un número para conocer la tabla de multiplicar: ")
		var input string
		fmt.Scanln(&input)
		num, err := strconv.Atoi(input)
		if err == nil {
			return num
		}
		fmt.Println("Error, por favor ingrese un número entero.")
	}
}

func mostrarTabla(num int) {
	fmt.Printf("\nTabla de multiplicar del %d:\n", num)
	for i := 1; i <= 10; i++ {
		fmt.Printf("%d x %d = %d\n", num, i, num*i)
	}
}